### PR TITLE
[Subgraph]Improve subgraph

### PR DESCRIPTION
**Pull request by Le-Caignec**

### CODE REVIEW

File: packages/subgraph/src/appRegistry.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+  // Create and save the protectedData entity
+  let app = App.load(contract._address.toHex());
+  if (!app) {
+    app = new App(contract._address.toHex());
+    app.addOnlyAppWhitelists = new Array<string>();
+  }
+  app.owner = contract.owner().toHex();
+  app.save();
+}
```
Comment (PierreJeanjacquot) [inline packages/subgraph/src/appRegistry.ts:19]: this will index all the apps on iExec and all the accounts owning an app, this is maybe not what we want.

**Review by PierreJeanjacquot — CHANGES_REQUESTED** [review]

File: packages/subgraph/src/appRegistry.ts (hunk 1 above)
Comment (Le-Caignec) [reply packages/subgraph/src/appRegistry.ts]: Commit: 0763f4087f1d2b5b13ed26862503e5aaebf0620c

File: packages/subgraph/src/addOnlyAppWhitelist.ts [hunk 2]
```diff
@@ -1,13 +1,24 @@
...
+import { App as AppContract } from '../generated/AppRegistry/App';
+import { checkAndCreateAccount } from './utils/utils';
 
 export function handleNewAppAdded(
   event: NewAppAddedToAddOnlyAppWhitelistEvent
 ): void {
+  let contract = AppContract.bind(event.params.appAddress);
+  checkAndCreateAccount(contract.owner().toHex());
+
   let app = App.load(event.params.appAddress.toHex());
```
[the remark this answers is not in the document]
Comment (Le-Caignec) [reply packages/subgraph/src/addOnlyAppWhitelist.ts:12]: i didn't find that very usefull

**Review by PierreJeanjacquot — APPROVED** [review]
LGTM

**Pull request merged**